optimism/src/keccak/witness.rs

```diff
@@ -302,15 +303,18 @@ impl<Fp: Field> KeccakInterpreter for KeccakEnv<Fp> {
         chi.state_f()
     }
 
-    fn run_iota(&mut self, state_f: &[u64], round: usize) {
+    fn run_iota(&mut self, state_f: &[u64], round: usize) -> Vec<u64> {
         let iota = Iota::create(state_f, round);
+        let state_g = iota.state_g();
 
         // Update columns
-        for i in 0..STATE_LEN {
-            self.write_column(KeccakColumn::IotaStateG(i), iota.state_g(i));
+        for (i, g) in state_g.iter().enumerate() {
```

Comment:
I suggest to add an additional assert that this value is equal to `STATE_LEN`.

Reply:
https://github.com/o1-labs/proof-systems/issues/1666